editor/backend/src/EditCommand.ts

```diff
@@ -31,23 +31,39 @@ export class EditCommand implements EditCommandIpc {
   public constructor(iModel: IModelDb, ..._args: any[]) {
     this.iModel = iModel;
   }
-  public get ctor(): EditCommandType { return this.constructor as EditCommandType; }
+  public get ctor(): EditCommandType {
+    return this.constructor as EditCommandType;
+  }
 
   public async onStart(): Promise<any> { }
 
   public async ping(): Promise<{ commandId: string, version: string, [propName: string]: any }> {
     return { version: this.ctor.version, commandId: this.ctor.commandId };
   }
 
-  public onCleanup(): void { }
+  // This is only temporary to find subclasses that used to implement this method. It was made async and renamed `requestFinish`.
+  private onFinish() { }
 
-  public onFinish(): void { }
+  /**
+   * Called when another EditCommand wishes to become the active EditCommand.
+   * Subclasses should complete and save their work as soon as possible and then return "done".
+   * If it is not currently possible to finish, return any string other than "done" and the other EditCommand will have to wait and retry,
+   * potentially showing the returned string to the user.
+   */
+  public async requestFinish(): Promise<"done" | string> {
+    this.onFinish(); // TODO: temporary, remove
```

Comment:
In this PR, or in a subsequent one?

Reply:
i'm worried about substation and synchro